src/scheduler/controller.rs

```diff
@@ -0,0 +1,102 @@
+//! The GC controller thread.
+
+use std::sync::mpsc::Receiver;
+use std::sync::Arc;
+
+use crate::plan::controller_collector_context::ControllerCollectorContext;
+use crate::scheduler::gc_work::ScheduleCollection;
+use crate::scheduler::CoordinatorMessage;
+use crate::util::VMWorkerThread;
+use crate::vm::VMBinding;
+use crate::MMTK;
+
+use super::{GCWorkScheduler, GCWorker};
+
+pub struct GCController<VM: VMBinding> {
+    mmtk: &'static MMTK<VM>,
+    requester: Arc<ControllerCollectorContext<VM>>,
+    scheduler: Arc<GCWorkScheduler<VM>>,
+    receiver: Receiver<CoordinatorMessage<VM>>,
+    coordinator_worker: Box<GCWorker<VM>>,
```

Comment:
Is there any reason that this needs to be boxed?

Reply:
The only reason is for consistency with GC worker threads which have `Box<GCWorker>` instances.  But it doesn't have to be boxed here in `GCCoordinator`.  I'll try to remove the `Box` here.